ERCS/erc-7595.md

```diff
@@ -0,0 +1,491 @@
+---
+eip: 7595
+title: Collateralized NFT
+description: ERC-721 Extension to enable collateralization with ERC-20 based tokens.
+author: 571nKY (@571nKY), Cosmos (@Cosmos4k), f4t50 (@f4t50), Harpocrates (@harpocrates555)
+discussions-to: https://ethereum-magicians.org/t/collateralized-nft-standard/18097
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-03-13
+requires: 20, 721
+---
+
+## Abstract
+
+This proposal recommends an extension of [EIP-721](./eip-721.md) to allow for collateralization using a list of [EIP-20](./eip-20.md) based tokens. The proprietor of this EIP collection could hold both the native coin and [EIP-20](./eip-20.md) based tokens, with the tokenId acting as the access key to unlock the associated portion of the underlying [EIP-20](./eip-20.md) balance.
+
+## Motivation
+
+“NFTfi” focuses on the NFT floor price to enable the market value of the NFT serve as a collateral in lending protocols. The NFT floor price is susceptible to the supply-demand dynamics of the NFT market, characterized by higher volatility compared to the broader crypto market. Furthermore, potential price manipulation in specific NFT collections can artificially inflate NFT market prices, impacting the floor price considered by lending protocols. Relying solely on the NFT floor price based on market value is both unpredictable and unreliable.
+
+This EIP addresses various challenges encountered by the crypto community with [EIP-721](./eip-721.md) based collections and assets. This EIP brings forth advantages such as sustainable NFT royalties supported by tangible assets, an on-chain verifiable floor price, and the introduction of additional monetization avenues for NFT collection creators.
+
+### Presets
+
+* The Basic Preset allows for the evaluation of an on-chain verifiable price floor for a specified NFT asset.
+
+* The Dynamic Preset facilitates on-chain modification of tokenURI based on predefined collateral rules for a specified NFT asset.
+
+* With the Royalty Preset, NFT collection creators can receive royalty payments for each transaction involving asset owners and Externally Owned Accounts (EOA), as well as transactions with smart contracts.
+
+* The VRF Preset enables the distribution of collateral among multiple NFT asset holders using the Verifiable Random Function (VRF) by Chainlink.
+
+### Extension to Existing EIP-721 Based Collections
+
+For numerous [EIP-721](./eip-721.md) based collections that cannot be redeployed, we propose the implementation of an abstraction layer embodied by a smart contract. This smart contract would replicate all the functionalities of this EIP standard and grant access to collateral through mapping.
```

Comment:
A wrapper contract could do, I think

Reply:
Wrapper contract is implemented for existing ERC-721 collections that would like to utilize the functionality of ERC-20 based collateral attached to the NFT collection without making too many changes to their NFT collection.

We also suggest a standard for ERC-721 creators who want additional customization, etc.